x-pack/plugins/ml/public/embeddables/lens/lens_vis_layer_selection_flyout/flyout_body.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useMemo } from 'react';
+import type { Embeddable } from '@kbn/lens-plugin/public';
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import type { LensPublicStart } from '@kbn/lens-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import './style.scss';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiTitle,
+  EuiSpacer,
+  EuiPanel,
+  EuiIcon,
+  EuiText,
+} from '@elastic/eui';
+
+import {
+  getLayers,
+  getJobsItemsFromEmbeddable,
+  convertLensToADJob,
+} from '../../../application/jobs/new_job/job_from_lens';
+import type { LayerResult } from '../../../application/jobs/new_job/job_from_lens';
+import { CREATED_BY_LABEL } from '../../../../common/constants/new_job';
+import { extractErrorMessage } from '../../../../common/util/errors';
+
+interface Props {
+  embeddable: Embeddable;
+  data: DataPublicPluginStart;
+  share: SharePluginStart;
+  lens: LensPublicStart;
+  onClose: () => void;
+}
+
+export const FlyoutBody: FC<Props> = ({ onClose, embeddable, data, share, lens }) => {
+  const embeddableItems = useMemo(() => getJobsItemsFromEmbeddable(embeddable), [embeddable]);
+
+  const [layerResult, setLayerResults] = useState<LayerResult[]>([]);
+
+  useEffect(() => {
+    const { vis } = embeddableItems;
+
+    getLayers(vis, data.dataViews, lens).then((layers) => {
+      setLayerResults(layers);
+    });
+  }, []);
+
+  function createADJob(layerIndex: number) {
+    convertLensToADJob(embeddable, share, layerIndex);
+  }
+
+  return (
+    <>
+      {layerResult.map((layer, i) => (
+        <>
+          <EuiPanel paddingSize="none">
+            <header className="mlLnsLayerPanel__layerHeader">
+              <EuiFlexGroup
+                gutterSize="s"
+                alignItems="center"
+                responsive={false}
+                className={'lnsLayerPanel__settingsStaticHeader'}
+              >
+                {layer.icon && (
+                  <EuiFlexItem grow={false}>
+                    <EuiIcon type={layer.icon} />
+                  </EuiFlexItem>
+                )}
+                <EuiFlexItem grow>
+                  <EuiTitle size="xxs">
```

Comment:
If we change this to `EuiText`, we can still use the heading tag, but also provide the color via the prop instead of applying a class on the element itself.